### PR TITLE
Issue #1499 - Use ingress host information to populate application external URL

### LINKED ISSUE #1499: External app links are not accurate
**Describe the bug**
Argo CD UI displays links for apps which have externally available IPs. Right now such links redirects directly to IP which don't make sense to Ingress. Instead of using IP UI should use ingress host name mentioned in ingress rule.

### PR DESCRIPTION
**Pull request by alexmt**
Closes #1499

### CODE REVIEW

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -216,16 +216,16 @@ type ApplicationDestination struct {
...
+	Resources      []ResourceStatus       `json:"resources,omitempty" protobuf:"bytes,1,opt,name=resources"`
+	Sync           SyncStatus             `json:"sync,omitempty" protobuf:"bytes,2,opt,name=sync"`
+	Health         HealthStatus           `json:"health,omitempty" protobuf:"bytes,3,opt,name=health"`
+	History        []RevisionHistory      `json:"history,omitempty" protobuf:"bytes,4,opt,name=history"`
+	Conditions     []ApplicationCondition `json:"conditions,omitempty" protobuf:"bytes,5,opt,name=conditions"`
+	ReconciledAt   metav1.Time            `json:"reconciledAt,omitempty" protobuf:"bytes,6,opt,name=reconciledAt"`
+	OperationState *OperationState        `json:"operationState,omitempty" protobuf:"bytes,7,opt,name=operationState"`
+	ObservedAt     metav1.Time            `json:"observedAt,omitempty" protobuf:"bytes,8,opt,name=observedAt"`
+	SourceType     ApplicationSourceType  `json:"sourceType,omitempty" protobuf:"bytes,9,opt,name=sourceType"`
+	BrowsableURLs  []string               `json:"browsableURLs,omitempty" protobuf:"bytes,10,opt,name=browsableURLs"`
```
Comment (alexec) [inline pkg/apis/application/v1alpha1/types.go]: Can we rename to `ExternalURLs` and document?

File: pkg/apis/application/v1alpha1/types.go [hunk 2]
```diff
@@ -524,10 +524,11 @@ type InfoItem struct {
...
 type ResourceNetworkingInfo struct {
-	TargetLabels map[string]string        `json:"targetLabels,omitempty" protobuf:"bytes,1,opt,name=targetLabels"`
-	TargetRefs   []ResourceRef            `json:"targetRefs,omitempty" protobuf:"bytes,2,opt,name=targetRefs"`
-	Labels       map[string]string        `json:"labels,omitempty" protobuf:"bytes,3,opt,name=labels"`
-	Ingress      []v1.LoadBalancerIngress `json:"ingress,omitempty" protobuf:"bytes,4,opt,name=ingress"`
+	TargetLabels  map[string]string        `json:"targetLabels,omitempty" protobuf:"bytes,1,opt,name=targetLabels"`
+	TargetRefs    []ResourceRef            `json:"targetRefs,omitempty" protobuf:"bytes,2,opt,name=targetRefs"`
+	Labels        map[string]string        `json:"labels,omitempty" protobuf:"bytes,3,opt,name=labels"`
+	Ingress       []v1.LoadBalancerIngress `json:"ingress,omitempty" protobuf:"bytes,4,opt,name=ingress"`
+	BrowsableURLs []string                 `json:"browsableURLs,omitempty" protobuf:"bytes,5,opt,name=browsableURLs"`
```
Comment (alexec) [inline pkg/apis/application/v1alpha1/types.go]: Can we rename to `ExternalURLs` and document?

**Review by alexec — APPROVED** [review]
Conditional approval.

File: pkg/apis/application/v1alpha1/types.go (hunk 1 above)
Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: Done

File: pkg/apis/application/v1alpha1/types.go (hunk 2 above)
Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: Done

**Pull request merged**